File: LeetCode/Explore/TopInterview/Amazon/ArrayString/16_3sum_closest.py
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        minDiff = 1000000
        nums.sort()
        
        for i in range(len(nums)):
          if minDiff != 0:
            l = i + 1
            r = len(nums) - 1
            while l < r:
              sum = nums[i] + nums[l] + nums[r]
              if abs(target - sum) < abs(minDiff):
                minDiff = target - sum
              if sum < target:
                l += 1
              else:
                r -= 1
              
        return target - minDiff

File: LeetCode/Explore/TopInterview/Amazon/ArrayString/16_3sum_closest.cpp
class Solution {
public:
    int threeSumClosest(vector<int>& nums, int target) {
      int minDiff = INT_MAX;
      sort(nums.begin(), nums.end());
      for (int i = 0; i < nums.size() && minDiff != 0; i++) {
        int l = i + 1, r = nums.size() - 1;
        while (l < r) {
          int sum = nums[i] + nums[l] + nums[r];
          if (abs(target - sum) < abs(minDiff)) {
            minDiff = target - sum;
          }
          if (sum - target < 0) {
            l++;
          }
          else {
            r--;
          }
        }
      }
      return target - minDiff;
    }
};
